### DISCUSSION TITLE
Patterns for testing code that uses CliWrap

### DISCUSSION THREAD

**Discussion by sfwester** [Tyrrrz/CliWrap#178] [Q&A] [answered]
Hi,

I was wondering if anybody has any patterns for testing code calling CliWrap? I haven't come up with anything too elegant myself and was wondering if I am missing something. The static class and the fluent design of the library make is hard to substitute in testing.

**Comment by Tyrrrz** [comment, marked as answer]
You are not supposed to substitute it. Either test your behaviors with it, or abstract it away at a higher level.

https://github.com/Tyrrrz/CliWrap/wiki/Migration-guide-(from-v2.5-to-v3.0)#removal-of-icli

**Reply by sfwester** [reply]
I can get behind that. Thanks!
